qubes/ext/gui.py

```diff
@@ -93,10 +91,8 @@ def on_domain_qdb_create(self, vm, event):
                 vm.untrusted_qdb.write('/qubes-gui-domain-xid',
                                        str(vm.guivm.xid))
 
-            # Add keyboard layout from that of GuiVM
-            kbd_layout = vm.guivm.features.get('keyboard-layout', None)
-            if kbd_layout:
-                vm.untrusted_qdb.write('/keyboard-layout', kbd_layout)
+        # Add keyboard layout
+        vm.untrusted_qdb.write('/keyboard-layout', vm.keyboard_layout)
```

Comment:
This (and other below) should be set only if all:
 - vm has some guivm
 - vm is not its own guivm
 - vm is running (it is already the case in this handler, but not necessary the others)

Otherwise a keyboard layout set via desktop environment tools in guivm (like xfce settings) would be overridden by this property (possibly reverting recent user change)